document/core/exec/instructions.rst

```diff
@@ -2625,8 +2625,21 @@ Control Instructions
 :math:`\TRY~\blocktype~\instr_1^\ast~(\CATCH~x~\instr_2^\ast)^\ast~(\CATCHALL~\instr_3^\ast)^?~\END`
 ....................................................................................................
 
-.. todo::
-   Add prose for the |TRY| - |CATCH| - |CATCHALL| execution step.
+1. Assert: due to :ref:`validation <valid-blocktype>`, :math:`\expand_F(\blocktype)` is defined.
+
+2. Let :math:`[t_1^n] \to [t_2^m]` be the :ref:`function type <syntax-functype>` :math:`\expand_F(\blocktype)`.
+
+3. Assert: due to :ref:`validation <valid-try-catch>`, there are at least :math:`n` values on the top of the stack.
+
+4. Pop the values :math:`\val^n` from the stack.
+
+5. Let :math:`L` be the label whose arity is :math:`m` and whose continuation is the end of the |TRY| instruction.
+
+6. Let :math:`H` be the :ref:`catching exception handler <syntax-handler>` :math:`\CATCHadm\{a_x~instr_2^\ast\}^\ast\{\epsilon~\instr_3^\ast\}^?`, whose mapping from tag addresses to branch targets corresponds to the |CATCH| and |CATCHALL| clauses, where :math:`a_x` is the tag address of each tag :math:`x`.
```

Comment:
@aheejin I'll change `a` to `a_x` in the equations as well, in the next commit. 

I agree that using the same word for the administrative `catch` is not very clear, but I seem to remember @rossberg thought it's clear from the context. FWIW the first `catch` is referred to as a [catching exception handler](https://webassembly.github.io/exception-handling/core/exec/runtime.html#exception-handlers) and the next `catch` as a clause of the `try-catch-catch_all` instruction. 

>  Should we prepend 'administrative instruction' to the first CATCHadm?

I meant to imply that the first `CATCHadm` is an administrative instruction by calling it a "catching exception handler". I'm not sure how to rephrase this to include "administrative instruction". Any suggestions?

> >  whose mapping from tag addresses to branch targets corresponds to the |CATCH| and |CATCHALL| clauses,
> What does this mean? What is the mapping from tag addresses to branch targets?

I was referring to the explanation given for [catching exception handlers](https://webassembly.github.io/exception-handling/core/exec/runtime.html#exception-handlers):

> Catching handlers start with the identifier and carry a mapping from tag addresses to their associated branch targets,

I'll try to think of a way to rephrase this, can't come up with something atm.

Reply:
> > > whose mapping from tag addresses to branch targets corresponds to the |CATCH| and |CATCHALL| clauses,
> > > What does this mean? What is the mapping from tag addresses to branch targets?
> 
> I was referring to the explanation given for [catching exception handlers](https://webassembly.github.io/exception-handling/core/exec/runtime.html#exception-handlers):

What 'branch targets' are we talking about here? I think this is the part I was confused about.

Reply:
The term "branch" here refers to a CFG of a try-catch instruction. Such a CFG has a central [basic block](https://en.wikipedia.org/wiki/Basic_block) checking the exception's tag, and from this block there are branches to all the catch blocks of the try-catch instruction. Depending on the exception's tag, say it's `a`, the execution flow follows the branch to the block of `catch a`.

Reply:
How about "catch bodies"? We don't use the term basic block in our formal spec, and we use the term "branch" only to mean `br`/`br_if`/`br_table` instructions, so I think it can be confusing.

Reply:
done